### PR TITLE
Added examples and more unit tests for multi constructor feature

### PR DESCRIPTION
**Pull request by benflexcompute**

### CODE REVIEW

File: tests/simulation/framework/test_multi_constructor_model.py [hunk 1]
```diff
@@ -47,18 +57,132 @@ def test_full_model(get_aerospace_condition_default, get_aerospace_condition_usi
...
+            "private_attribute_constructor",
+            "private_attribute_input_cache",
+        ]:
+            incomplete_entity[key] = value
+    incomplete_data["entities"]["stored_entities"].append(incomplete_entity)
+    incomplete_data["entities"]["stored_entities"].append(
+        full_data["entities"]["stored_entities"][2]
+    )
+    # with open("boxes.json", "w") as fp:
+    #     json.dump(incomplete_data, fp, indent=4)
```
Comment (maciej-flexcompute) [inline tests/simulation/framework/test_multi_constructor_model.py:186]: remove

File: tests/simulation/framework/test_multi_constructor_model.py [hunk 2]
```diff
@@ -47,18 +57,132 @@ def test_full_model(get_aerospace_condition_default, get_aerospace_condition_usi
...
+        "private_attribute_constructor": full_data["thermal_state"][
+            "private_attribute_constructor"
+        ],
+        "private_attribute_input_cache": full_data["thermal_state"][
+            "private_attribute_input_cache"
+        ],
+    }
+
+    # with open("default_and_from.json", "w") as fp:
+    #     json.dump(incomplete_data, fp, indent=4)
```
Comment (maciej-flexcompute) [inline tests/simulation/framework/test_multi_constructor_model.py:102]: remove

File: tests/simulation/framework/test_multi_constructor_model.py [hunk 3]
```diff
@@ -47,18 +57,132 @@ def test_full_model(get_aerospace_condition_default, get_aerospace_condition_usi
...
+
+    full_data = get_aerospace_condition_using_from.model_dump(exclude_none=True)
+    incomplete_data = {
+        "type_name": full_data["type_name"],
+        "private_attribute_constructor": full_data["private_attribute_constructor"],
+        "private_attribute_input_cache": full_data["private_attribute_input_cache"],
+    }
+
+    # with open("incomplete_data.json", "w") as fp:
+    #     json.dump(incomplete_data, fp, indent=4)
```
Comment (maciej-flexcompute) [inline tests/simulation/framework/test_multi_constructor_model.py:82]: remove

File: tests/simulation/framework/test_multi_constructor_model.py [hunk 4]
```diff
@@ -47,18 +57,132 @@ def test_full_model(get_aerospace_condition_default, get_aerospace_condition_usi
...
 
-def test_incomplete_model(get_aerospace_condition_default, get_aerospace_condition_using_from):
+def test_incomplete_model(
+    get_aerospace_condition_default,
+    get_aerospace_condition_using_from,
+    get_aerospace_condition_default_and_thermal_state_using_from,
+):
     full_data = get_aerospace_condition_default.model_dump(exclude_none=True)
+    # with open("using_default_constructor.json", "w") as fp:
+    #     json.dump(full_data, fp, indent=4)
```
Comment (maciej-flexcompute) [inline tests/simulation/framework/test_multi_constructor_model.py:67]: remove

File: tools/integrations/schema_generation_v2.py [hunk 5]
```diff
@@ -673,3 +683,115 @@ def write_example(
...
+write_example(example_dict, "multi_constructor", "default_and_from_constructor.json")
+
+
+# 4. entity list of boxes:
+example_dict = {
+    "entities": {
+        "stored_entities": [
+            {
+                "private_attribute_registry_bucket_name": "VolumetricEntityType",
+                "private_attribute_entity_type_name": "Box",
```
Comment (maciej-flexcompute) [inline tools/integrations/schema_generation_v2.py:763]: if that is for integration demonstration, do we need these?

File: tools/integrations/schema_generation_v2.py [hunk 6]
```diff
@@ -673,3 +683,115 @@ def write_example(
...
+                "name": "my_box_default",
+                "private_attribute_zone_boundary_names": {"items": []},
```
Comment (maciej-flexcompute) [inline tools/integrations/schema_generation_v2.py:765]: if that is for integration demonstration, do we need this?

File: tools/integrations/schema_generation_v2.py [hunk 7]
```diff
@@ -673,3 +683,115 @@ def write_example(
...
+                    "center": {"value": [7.0, 1.0, 2.0], "units": "m"},
+                    "size": {"value": [2.0, 2.0, 3.0], "units": "m"},
+                    "name": "my_box_from",
+                },
+            },
+            {
+                "private_attribute_registry_bucket_name": "VolumetricEntityType",
+                "private_attribute_entity_type_name": "Cylinder",
+                "name": "my_cylinder_default",
+                "private_attribute_zone_boundary_names": {"items": []},
```
Comment (maciej-flexcompute) [inline tools/integrations/schema_generation_v2.py:788]: same here

**Comment by benflexcompute** [comment]
REPLACED BY https://github.com/flexcompute/Flow360/pull/409

**Pull request closed without merging**